src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1851 @@ interface MPSMatrixMultiplication {
+
+		[Export ("destinationsAreTransposed")]
+		bool DestinationsAreTransposed { get; }
+
+		[Export ("encodeToCommandBuffer:copyDescriptor:")]
+		void Encode (IMTLCommandBuffer cmdBuf, MPSMatrixCopyDescriptor copyDescriptor);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	interface MPSImageCopyToMatrix {
+		[Export ("destinationMatrixOrigin", ArgumentSemantic.Assign)]
+		MTLOrigin DestinationMatrixOrigin { get; set; }
+
+		[Export ("destinationMatrixBatchIndex")]
+		nuint DestinationMatrixBatchIndex { get; set; }
+
+		[Export ("dataLayout")]
+		MPSDataLayout DataLayout { get; }
+
+		[Export ("initWithDevice:dataLayout:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSDataLayout dataLayout);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:sourceImage:destinationMatrix:")]
+		void Encode (IMTLCommandBuffer commandBuffer, MPSImage sourceImage, MPSMatrix destinationMatrix);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	interface MPSImageFindKeypoints {
+		[Export ("keypointRangeInfo")]
+		MPSImageKeypointRangeInfo KeypointRangeInfo { get; }
+
+		[Export ("initWithDevice:info:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSImageKeypointRangeInfo info);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:sourceTexture:regions:numberOfRegions:keypointCountBuffer:keypointCountBufferOffset:keypointDataBuffer:keypointDataBufferOffset:")]
+		void Encode (IMTLCommandBuffer commandBuffer, IMTLTexture source, MTLRegion regions, nuint numberOfRegions, IMTLBuffer keypointCountBuffer, nuint keypointCountBufferOffset, IMTLBuffer keypointDataBuffer, nuint keypointDataBufferOffset);
+	}
+	
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSBinaryImageKernel))]
+	interface MPSImageArithmetic {
+		[Export ("primaryScale")]
+		float PrimaryScale { get; set; }
+
+		[Export ("secondaryScale")]
+		float SecondaryScale { get; set; }
+
+		[Export ("bias")]
+		float Bias { get; set; }
+
+		[Export ("primaryStrideInPixels", ArgumentSemantic.Assign)]
+		MTLSize PrimaryStrideInPixels { get; set; }
+
+		[Export ("secondaryStrideInPixels", ArgumentSemantic.Assign)]
+		MTLSize SecondaryStrideInPixels { get; set; }
+
+		//inlining ctor from base class
+		[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSImageArithmetic))]
```

Comment:
DisableDefaultCtor

Reply:
The MPSImageArithmetic subclasses are all able to be created using 'init' in an Xcode project without errors.